packages/xforms-engine/src/parse/model/SecondaryInstance/sources/ExternalSecondaryInstanceResource.ts

```diff
@@ -0,0 +1,222 @@
+import type { JRResourceURL } from '@getodk/common/jr-resources/JRResourceURL.ts';
+import type { MissingResourceBehavior } from '../../../../client/constants.ts';
+import type { FetchResource, FetchResourceResponse } from '../../../../client/resources.ts';
+import { ErrorProductionDesignPendingError } from '../../../../error/ErrorProductionDesignPendingError.ts';
+import { FormAttachmentResource } from '../../../attachments/FormAttachmentResource.ts';
+import type { ExternalSecondaryInstanceSourceFormat } from './SecondaryInstanceSource.ts';
+
+const assertResponseSuccess = (resourceURL: JRResourceURL, response: FetchResourceResponse) => {
+	const { ok = true, status = 200 } = response;
+
+	if (!ok || status !== 200) {
+		throw new ErrorProductionDesignPendingError(`Failed to load ${resourceURL.href}`);
+	}
+};
+
+const stripContentTypeCharset = (contentType: string): string => {
+	return contentType.replace(/;charset=.*$/, '');
```

Comment:
What do you think about this:

```diff
- return contentType.replace(/;charset=.*$/, '');
+ return contentType.replace(/;\s?charset=.*$/, '');
```

Sometimes response contenttype includes space between media type and charset, including examples here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#examples

Reply:
Can you file an issue? This PR has already been merged, so I can't make changes here.